### DISCUSSION TITLE
UNION processing

### DISCUSSION THREAD

**Discussion by cptb** [bruce-dunwiddie/tsql-parser#80] [Q&A] [answered]
Hi Bruce,
Thanks for the amazing tool and the answer you provided for my last question.
I am facing another issue right now and it is to deal with the UNION. See this example:

SELECT 1
UNION
SELECT 2
UNION
SELECT 3

The current version chunks the code into two sections and removes the second UNION. So when I execute it, it produced two result sets rather than just 1.

Is there a setting anywhere that the whole UNIONs code to be kept in a single section? I have some SQL code provided by others that contains 10-20 UNIONs all together. The produced multiple result sets at the moment.

I appreciate your comments, and thank you again.

Here is the output I feed the above SQL into the parser.

Start...
Statement 1:  select 1 union select 2
        Token type: Keyword     Text: select
        Token type: NumericLiteral      Text: 1
        Token type: Keyword     Text: union
        Token type: Keyword     Text: select
        Token type: NumericLiteral      Text: 2

Statement 2:  select 3
        Token type: Keyword     Text: select
        Token type: NumericLiteral      Text: 3

Regards
Paul

**Comment by bruce-dunwiddie** [comment]
Thank you for posting this. I'll look into it and get back to you.

**Comment by bruce-dunwiddie** [comment]
I've verified that it's not working as I would expect, so created an issue, https://github.com/bruce-dunwiddie/tsql-parser/issues/81 .

I've got a large amount of changes sitting in the dev branch, so I'll work on how to resolve as well as see what of those changes can be released versus left out for now.

**Comment by bruce-dunwiddie** [comment, marked as answer]
Resolved in latest release, https://www.nuget.org/packages/TSQL.Parser/2.0.0 .

Example syntax, https://dotnetfiddle.net/kL5Uxw .
